### ISSUE TITLE
Run the code with different image sizes

### ISSUE THREAD

**Issue by aminpdik** [ZrrSkywalker/Personalize-SAM#17]
Thanks for the great work.
@jiangzhengkai @ZrrSkywalker 

I am gonna try the code with different image sizes. But I get mismatch error. Is there any way to edit the code for sizes other than 1024 * 1024?

**Comment by jiangzhengkai** [comment]
@aminpdik You can resize the rest image size to the first one, there is no need to resize all the images to 1024 * 1024 size.

**Comment by aminpdik** [comment]
Thanks for your response. 

I am gonna try SAM without finetuning. When I try images with size 400 by 400, I get an error. So, I changed the input size to build sam with size 400 by 400, but I get mismatch error to load weights for SAM.

**Comment by ZrrSkywalker** [comment]
@aminpdik Hi, our PerSAM and the original SAM can take as input any image resolutions, and will inherently conduct a pre-resize operation before the image encoder.

**Comment by aminpdik** [comment]
Yes It can take. But the provided weight for different SAM structure like ViT-H only works for input size 1024 by 1024. When I change the size, it returns mismatch error when load weights. 

Could you please advise me with this issue.

**Comment by ZrrSkywalker** [comment]
@aminpdik Yes, the size of SAM itself cannot be simply changed, since it has learned some positional encodings with fixed size.

**Comment by aminpdik** [comment]
@ZrrSkywalker I am gonna use SAM image encoder as a feature extraction. Do you have any idea to edit SAM to work with different sizes?

**Comment by ZrrSkywalker** [comment]
@aminpdik You may first load the model with 1024x1024 resolutions and then resize the learnable pos embed and other related modules in the image encoder into new sizes.
